sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -640,11 +641,20 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
    * Create a [[DropTableCommand]] command.
    */
   override def visitDropTable(ctx: DropTableContext): LogicalPlan = withOrigin(ctx) {
-    DropTableCommand(
-      visitTableIdentifier(ctx.tableIdentifier),
-      ctx.EXISTS != null,
-      ctx.VIEW != null,
-      ctx.PURGE != null)
+    visitMultipartIdentifier(ctx.multipartIdentifier()) match {
```

Comment:
The way we do this for `CreateTable` is to parse to an intermediate representation, `CreateTableStatement` that contains exactly what was parsed. Then translation to v1 `DropTableCommand` or v2 `DropTable` can be done in an analyzer rule. That way, the parser and the analyzer stay separate and the parser implementation for `DROP TABLE` can be kept in Catalyst.

I think this should take the same approach.